doc/source/whatsnew/v1.0.0.rst

```diff
@@ -581,6 +581,7 @@ Deprecations
 - The ``pandas.util.testing`` module has been deprecated. Use the public API in ``pandas.testing`` documented at :ref:`api.general.testing` (:issue:`16232`).
 - ``pandas.SparseArray`` has been deprecated.  Use ``pandas.arrays.SparseArray`` (:class:`arrays.SparseArray`) instead. (:issue:`30642`)
 - The parameter ``is_copy`` of :meth:`DataFrame.take` has been deprecated and will be removed in a future version. (:issue:`27357`)
+- Support for indexing indexing an :class:`Index` with ``index[:, None]`` is deprecated and will be removed in a future version (:issue:`30588`)
```

Comment:
```suggestion
- Support for indexing an :class:`Index` with ``index[:, None]`` is deprecated and will be removed in a future version (:issue:`30588`)
```

Reply:
Also, it is not only about `index[:, None]` but about multi-dimensional indexing in general (but where `index[:, None]`  is a good example). So should write this more general. 
The recommended alternative is to convert to a numpy array explicitly first, before indexing (I would add that)